3D Chess/Assets/Scripts/PieceEntity.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Realms;
+using UnityEngine;
+
+public class PieceEntity : RealmObject
+{
+    public PieceType PieceType
+    {
+        get { return (PieceType)Type; }
+        set { Type = (int)value; }
+    }
```

Comment:
This can be an expression property:

```suggestion
    public PieceType PieceType
    {
        get => (PieceType)Type;
        set => Type = (int)value;
    }
```